bin/utest/vm_map_prot.c

```diff
@@ -83,68 +54,40 @@ int test_vmmap_data_x(void) {
 
   func_int_t ff = (func_int_t)func_buf;
 
-  if (sigsetjmp(return_to, 1) == 0) {
+  siginfo_t si;
+  TEST_EXPECT_SIGNAL(SIGSEGV, &si, ff, SEGV_ACCERR) {
     /* Executing function in data segment (without exec prot) */
     int v = ff(1);
     assert(v != 2);
   }
-
-  printf("SIGSEGV address: %p\nSIGSEGV code: %d\n", sigsegv_address,
-         sigsegv_code);
-
-  assert(sigsegv_address == ff);
-  assert(sigsegv_code == SEGV_ACCERR);
-
-  signal(SIGSEGV, SIG_DFL);
+  printf("SIGSEGV address: %p\nSIGSEGV code: %d\n", si.si_addr, si.si_code);
   return 0;
 }
 
-static const char *ro_data_str = "String in .rodata section";
+/* String must be alinged as instuctions because we jump here */
+static __aligned(4) const char ro_data_str[] = "String in .rodata section";
 
-int test_vmmap_rodata_w(void) {
-  setup_sigaction();
+int test_vmmap_rodata_access(void) {
+  siginfo_t si;
 
+  /* Check write */
   volatile char *c = (char *)ro_data_str;
-
   printf("mem to write: 0x%p\n", c);
-
-  if (sigsetjmp(return_to, 1) == 0) {
+  TEST_EXPECT_SIGNAL(SIGSEGV, &si, c, SEGV_ACCERR) {
     /* Writing to rodata segment. */
     *c = 'X';
     assert(*c == 'X');
   }
+  printf("SIGSEGV address: %p\nSIGSEGV code: %d\n", si.si_addr, si.si_code);
 
-  printf("SIGSEGV address: %p\nSIGSEGV code: %d\n", sigsegv_address,
-         sigsegv_code);
-
-  assert(sigsegv_address == c);
-  assert(sigsegv_code == SEGV_ACCERR);
-
-  signal(SIGSEGV, SIG_DFL);
-  return 0;
-}
-
-int test_vmmap_rodata_x(void) {
-  setup_sigaction();
-
+  /* Check execute */
   func_int_t fun = (func_int_t)ro_data_str;
-
   printf("func: 0x%p\n", fun);
-
-  if (sigsetjmp(return_to, 1) == 0) {
+  TEST_EXPECT_SIGNAL(SIGSEGV, &si, fun, SEGV_ACCERR) {
     /* Executing from rodata segment. */
     fun(1);
     assert(0);
```

Comment:
Why `assert(0)` here (and in many other places)? Is it possible to move it to `TEST_EXPECT_SIGNAL` macro, so the loop will perform `exit(1)` if the program has not received the signal?